roaring/roaring.go

```diff
@@ -2137,15 +2138,95 @@ func intersectArrayBitmap(a, b *Container) *Container {
 	return output
 }
 
-func intersectBitmapBitmap(a, b *Container) *Container {
-	output := &Container{bitmap: make([]uint64, bitmapN), containerType: containerBitmap}
+var bitmapAnd func(a, b, c []uint64) int
+var bitmapOr func(a, b, c []uint64) int
+var bitmapXor func(a, b, c []uint64) int
+var bitmapAndN func(a, b, c []uint64) int
+
+func goAnd(a, b, c []uint64) int {
+	// local variables added to prevent BCE checks in loop
+	// see https://go101.org/article/bounds-check-elimination.html
+	var (
+		ab = a[:bitmapN]
+		bb = b[:bitmapN]
+		ob = c[:bitmapN]
+		n  int
+	)
+	for i := 0; i < bitmapN; i++ {
+		v := ab[i] & bb[i]
+		ob[i] = v
```

Comment:
why not set ob[i] directly? `ob[i] = ab[i] & bb[i]`

Reply:
because i use the stored value to get the bitcount on the next line